FreeRTOS/FreeRTOS-Kernel
# GCC/ATmegaxxxx port path and name change.

Description
-----------
Per discussions offline and in PR #48, moving ```portable/GCC/ATmegaxxxx``` to ```portable/ThirdParty/GCC/ATmega```. The plan is to mature the port there, and when we are done move back. 

Things we are hoping improve:
- Default tick timer, weak linkage or extern linkage.
- Test on various devices/smulator port. So far the tests are done on Atmel Studio Simulator, limited to ATmega2560. 8-bit timer configuration/resolution and task context push/pop are checked. Watchdog timer is briefly tested. The rest needs more testing. Reference: https://github.com/FreeRTOS/FreeRTOS/pull/55

Test Steps
-----------
N/A. Since:
- Source code is not changed.
- There's no existing project in FreeRTOS/FreeRTOS repo, so no project is impacted by this update. 

Related Issue
-----------
N/A


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
